src/Keboola/S3Extractor/S3AsyncDownloader.php

```diff
@@ -105,6 +105,7 @@ private function makeCommandPool(): CommandPool
                     gc_collect_cycles();
                 }
                 $this->processFulfilled($result, $index);
+                unset($this->filesParameter[$index]);
```

Comment:
Pre `retry` sa tu ukladaju `parameters`, to bol dalsie `memory leak`, po tom ako sa subor uspesne stiahne sa moze dane `array` s parameters uvolnit.